### ISSUE TITLE
Can't change stepsize

### ISSUE THREAD

**Issue by JdeH** [Michael-F-Ellis/NearlyPurePythonWebAppDemo#5]
Hi Michael,

In order to include the newest version of your demo in Transcrypt, I've been playing with it a bit.
I've encountered 2 problems:

- Whenever I change the step size, it jumps back to its old value (most of the time).
- File server.py now holds a line saying `import client`. Is that really right?

No hurry, first enjoy being home again...

KR
Jacques

**Comment by Michael-F-Ellis** [comment]
Thanks, Jacques. I'm enjoying Paris at the moment :-)

I'll take a  look at the step size issue.

The import of client in server.py is intended. It's done to take advantage of Bottle's reloader mechanism which monitors for changes in all the files it can find in sys.modules.  I should probably add an explanatory comment above the import.

**Comment by JdeH** [comment]
While you're in Paris, if you find the time, perhaps visit Musee Rodin... (if you haven't already done so) My wife and I liked it a lot!

**Comment by Michael-F-Ellis** [comment]
The step size issue is fixed in the new single-file version of the demo, `allinone.py`.  It also no longer has to import client.py since all the code is now in the same module.  

**Explanation**
The step size problem was caused by the use of a text input with a submit button.  The code updates it with the server's current value for step size whenever the text input loses focus.  In client.py there's a handler bound to Enter that causes a submit.  So if you entered a value and hit Enter, the input never lost focus and no problem. But if you tab over to the submit button the input loses focus and the value you typed in can get erased at the next server update.

In the interest of keeping the demo code simple, I opted to replace the text input and submit button with a range slider. The slider fires a 'change' event when you release the button and I've bound that to the handler that submits the new values.

I'm leaving this issue open for now until I decide whether or not to update the older client.py file.
